stratum/hal/lib/common/hal.cc

```diff
@@ -247,7 +247,23 @@ void Hal::HandleSignal(int value) {
   // with no deadline will block forever, as it waits for all the active RPCs
   // to finish. To fix this, we give a deadline set to "now" so the call returns
   // immediately.
-  external_server_->Shutdown(std::chrono::system_clock::now());
+  // Per the grpc::Server docs, some other thread must call Shutdown() for Wait()
+  // to ever return. So, we try to call shutdown it a separate thread.
+  pthread_t shutdown_tid;
+  int ret = pthread_create(&shutdown_tid, nullptr,
```

Comment:
The list of functions safe to call from a signal handler is quite short (`man signal-safety`). `pthread_create` is not one of them :(

Reply:
I guess this makes sense... this probably means that we need a fix inside of gRPC?